### PR TITLE
DRAFT: points snap

### PR DESCRIPTION
**Pull request by matepal00**

### CODE REVIEW

File: eclipse/src/points-snap.ts [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+} from "@invariant-labs/sdk-eclipse";
+import {
+  ConfirmedSignatureInfo,
+  Connection,
+  ParsedTransactionWithMeta,
+  PublicKey,
+} from "@solana/web3.js";
+import * as fs from "fs";
+
+enum EventTypes {
```
Comment (Sniezka1927) [inline eclipse/src/points-snap.ts:20]: There is export from invariant package `InvariantEventNames`

File: eclipse/src/points-snap.ts [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+      "confirmed"
+    );
+
+    if (signatures.length === 0) {
+      done = true;
+      break;
+    }
+
+    allSignatures.push(...signatures);
+    if (lastTxHash === undefined) {
```
Comment (Sniezka1927) [inline eclipse/src/points-snap.ts:47]: Is this check needed?

Comment (Sniezka1927) [inline eclipse/src/points-snap.ts:47]: Is this check needed?

File: eclipse/src/points-snap.ts [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+        );
+        return processParsedTransactions(
+          await connection.getParsedTransactions(batchSignatures, "confirmed")
+        );
+      })
+    )
+  ).flat();
+};
+
+const extractEvents = (
```
Comment (Sniezka1927) [inline eclipse/src/points-snap.ts:94]: After parsing an event you should check program id that emitted event

File: eclipse/src/points-snap.ts [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+        break;
+      default:
+        return;
+    }
+  });
+  return eventsObject;
+};
+
+export const createPointsSnap = async (network: Network) => {
+  const MAX_SIGNATURES_PER_CALL = 300;
```
Comment (Sniezka1927) [inline eclipse/src/points-snap.ts:125]: That's the config variable. You can put it at the top of the file

Comment (Sniezka1927) [inline eclipse/src/points-snap.ts:125]: That's the config variable. You can put it at the top of the file

File: eclipse/src/points-snap.ts [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+  fs.writeFileSync(lastTxHashFileName, JSON.stringify(data, null, 2));
+  const txLogs = await fetchTransactionLogs(
+    connection,
+    sigs,
+    MAX_SIGNATURES_PER_CALL
+  );
+
+  const finalLogs = txLogs.flat();
+  const initialEvents = JSON.parse(
+    fs.readFileSync(lastTxHashFileName, "utf-8")
```
Comment (Sniezka1927) [inline eclipse/src/points-snap.ts:176]: Path to file does not look correct

**Review by Sniezka1927 — CHANGES_REQUESTED** [review]
The logic could be splitted into separated files to make the script cleaner. For example you can introduce `consts.ts` and `utils.ts`

File: eclipse/src/points-snap.ts (hunk 2 above)
Comment (matepal00) [reply eclipse/src/points-snap.ts:47]: only in the first iteration when there is not tx hash but yes

File: eclipse/src/points-snap.ts (hunk 3 above)
Comment (matepal00) [reply eclipse/src/points-snap.ts:94]: we gather tx hashes from program id, and then analyze program data, so it is event needed?

**Pull request closed without merging**